src/curve_fit.jl

```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
 end
 
 #for inplace f and inplace g
-function lmfit(f!, g!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; kwargs...)
-    R = OnceDifferentiable(f!, g!, p0, similar(r); inplace = true)
+function lmfit(f, g, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, kwargs...)
+    R = OnceDifferentiable(f, g, p0, similar(r); inplace = true)
     lmfit(R, p0, wt; kwargs...)
 end
 
 #for inplace f only
-function lmfit(f!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; autodiff = :finite, kwargs...)
-    autodiff = autodiff == :forwarddiff ? :forward : autodiff
-    R = OnceDifferentiable(f!, p0, similar(r); inplace = true, autodiff = autodiff)
+function lmfit(f, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, autodiff = :finite, kwargs...)
+    R = OnceDifferentiable(f, p0, similar(r); inplace = true, autodiff = autodiff)
     lmfit(R, p0, wt; kwargs...)
 end
 
+#experimental geo, I let the `inplace` machinery but it isn't avilable yet
+function lmfit(f, g, avv!, p0::AbstractArray, wt::AbstractArray; inplace = false, kwargs...)
```

Comment:
This function is actually not used anymore, I forgot to remove it, sorry! `avv!` is now passed as a `kwarg`.